servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpConnectionContext.java

```diff
@@ -27,4 +27,9 @@ public interface HttpConnectionContext extends ConnectionContext {
 
     @Override
     HttpProtocolVersion protocol();
+
+    // Do not override the return type of parent() method to HttpConnectionContext as there is a possibility to have an
+    // HttpConnectionContext over a non-HTTP ConnectionContext. Also, with HttpConnectionContext as a return type it
+    // makes implementation a lot harder because we receive a non-HTTP NettyConnectionContext from the transport and
+    // wrap it later.
```

Comment:
Note that we can make this change only now or never, as even adding a `default` impl will be a breaking change. Users who already implement `parent()` returning `ConnectionContext` type will be broken if we change it to `HttpConnectionContext` later.

Reply:
However, we can always add something like `HttpConnectionContext  httpParent()` later, if there is a use-case for that.

Reply:
The covariant return override should only be used if it can be known for certain that the return type will be the override type. If there are no certain cases, that's OK.